### DISCUSSION TITLE
How to include all packages from the workspace into tsup bundle?

### DISCUSSION THREAD

**Discussion by VanishMax** [egoist/tsup#613] [Q&A]
I'm not sure what causes my problem, so I'd like to raise an issue here:

In my monorepo (with pnpm workspaces, CJS modules), I have a couple of packages built with `tsup`, and all of them are used in the the `cli` package like this:

```json
"scripts": {
    "build": "tsup index.ts --dts-resolve"
  },
  "bin": {
    "cli": "dist/index.js"
  },
  "devDependencies": {
    "first": "workspace:*",
    "second": "workspace:*"
  },
```

As you can see, I use workspace packages in the CLI, bundle them with `tsup` and create a `bin` out of it. The problem becomes clear when I create a project to test the CLI:

```json
{
  "scripts": {
    "start": "cli"
  },
  "dependencies": {
    "cli": "0.1.0",
    "first": "0.1.0",
    "second": "0.1.0"
  }
}
```

If I include all worspace packages into dependencies, then the `cli` script runs perfectly. But without these dependencies, it raises an issue that some code is unavailable. It got me thinking: does **tsup** actually include workspace dependencies inside its bundle?

**Comment by egoist** [comment]
they should be bundled, otherwise it seems like a bug with workspace but that never happens to me. a minimal repro is welcome
